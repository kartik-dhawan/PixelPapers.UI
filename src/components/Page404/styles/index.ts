import { GLOBAL_COLORS, PAGE_LEVEL_COLORS } from "@/utils/colorSchemes"

export const styles = {
  notFoundWrapper: {
    height: "100svh",
    justifyContent: "center",
    alignItems: "center",
    padding: "2rem",
    boxSizing: "border-box",
    color: GLOBAL_COLORS.TEXT_PRIMARY_DARK,
    backgroundColor: PAGE_LEVEL_COLORS.PAGE_404.BACKGROUND_PRIMARY_LIGHT,
  },
  notFoundStatusCode: {
    fontSize: {
      xs: "72px",
      md: "80px",
    },
    margin: "1rem 0rem",
    color: PAGE_LEVEL_COLORS.PAGE_404.TEXT_PRIMARY_DARK,
  },
  notFoundBarCode: {
    fontSize: {
      xs: "80px",
      md: "104px",
      xl: "120px",
    },
    textAlign: "center",
  },
  notFoundTextBox: {
    textAlign: "center",
    fontSize: {
      xs: "22px",
      xl: "24px",
    },
    letterSpacing: "-1px",
  },
  notFoundGoBackLink: {
    margin: "2rem 0rem",
    "& a": {
      textTransform: "uppercase",
      position: "relative",
      fontSize: {
        xs: "18px",
        md: "22px",
        xl: "26px",
      },
      fontWeight: 700,
      color: GLOBAL_COLORS.TEXT_PRIMARY_DARK,
      borderBottom: "1.5px solid #6b757e",
      textDecoration: "none",
      "&::before": {
        content: `""`, // eslint-disable-line
        position: "absolute",
        height: "2px",
        width: "0%",
        backgroundColor: GLOBAL_COLORS.TEXT_PRIMARY_DARK,
        bottom: "-2px",
        left: 0,
        transition: "200ms all ease",
      },
      "&:hover::before": {
        width: "100%",
      },
    },
  },
}
