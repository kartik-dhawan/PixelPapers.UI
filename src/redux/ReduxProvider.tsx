"use client"

/**
 *
 * This component acts as a parent for client side components that require redux store
 * Wrap the component with this 'Redux Provider' to use state management on client side
 *
 * the updates made on the client side in the store will not be reflected in the server side components
 *
 * SERVER SIDE:
 * - Dispatching: store.dispatch(functionName(payload))
 * - Getting data from store: store.getState().sliaceName.stateName
 *
 * CLIENT SIDE:
 * - Dispatching: dispatch(functionName(payload)) [useDispatch()]
 * - Getting data from store: useSelector((state)=>state.slice.statename)
 *
 * Ref: https://www.youtube.com/watch?v=dRLjoT4r-jc
 */

import { Provider } from "react-redux"
import { store } from "./store"
import { ChildrenProps } from "@/utils/interfaces"

const ReduxProvider = ({ children }: ChildrenProps) => {
  return <Provider store={store}>{children}</Provider>
}

export default ReduxProvider
