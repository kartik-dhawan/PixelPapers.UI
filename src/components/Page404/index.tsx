import { dmSans, barcode, anton } from "@/utils/fonts"
import { Box, Stack } from "@mui/material"
import Link from "next/link"
import { styles } from "./styles"

const Page404 = () => {
  const nid = "notFound"

  return (
    <Stack
      className={nid + "Wrapper " + dmSans.className}
      data-testid={nid + "Wrapper"}
      sx={styles.notFoundWrapper}
    >
      <Box
        className={nid + "StatusCode " + anton.className}
        data-testid={nid + "StatusCode"}
        component={"h1"}
        sx={styles.notFoundStatusCode}
      >
        404
      </Box>
      <Box
        className={nid + "BarCode " + barcode.className}
        data-testid={nid + "BarCode"}
        sx={styles.notFoundBarCode}
      >
        R u lost, babygirl?
      </Box>
      <Box
        className={nid + "Textbox"}
        data-testid={nid + "Textbox"}
        sx={styles.notFoundTextBox}
      >
        Don&lsquo;t understand bar code? Yeah we don&lsquo;t understand your
        request either.
        <br />
        Please enter a valid URL.
      </Box>
      <Box
        className={nid + "GoBackLink"}
        data-testid={nid + "GoBackLink"}
        sx={styles.notFoundGoBackLink}
      >
        <Link
          href="/"
          aria-label="Link that will take you back to the home page of this blog wesbite."
        >
          Want us to take you back &lsquo;home&lsquo;?
        </Link>
      </Box>
    </Stack>
  )
}

export default Page404
