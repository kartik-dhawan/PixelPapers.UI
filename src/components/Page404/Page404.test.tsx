import ReduxProvider from "@/redux/ReduxProvider"
import "@testing-library/jest-dom"
import renderer from "react-test-renderer"
import Page404 from "."

describe("The 404 Page component", () => {
  it("-- should match the snapshot", () => {
    const tree = renderer.create(
      <ReduxProvider>
        <Page404 />
      </ReduxProvider>,
    )

    expect(tree).toMatchSnapshot()
  })
})
