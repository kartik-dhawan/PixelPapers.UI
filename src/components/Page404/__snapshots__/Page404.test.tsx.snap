// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The 404 Page component -- should match the snapshot 1`] = `
<div
  className="MuiStack-root notFoundWrapper className css-1fuvqsc-MuiStack-root"
  data-testid="notFoundWrapper"
>
  <h1
    className="notFoundStatusCode className MuiBox-root css-1ju47en"
    data-testid="notFoundStatusCode"
  >
    404
  </h1>
  <div
    className="notFoundBarCode className MuiBox-root css-1xmcpmd"
    data-testid="notFoundBarCode"
  >
    R u lost, babygirl?
  </div>
  <div
    className="notFoundTextbox MuiBox-root css-1dnlse9"
    data-testid="notFoundTextbox"
  >
    Don‘t understand bar code? Yeah we don‘t understand your request either.
    <br />
    Please enter a valid URL.
  </div>
  <div
    className="notFoundGoBackLink MuiBox-root css-14yzx88"
    data-testid="notFoundGoBackLink"
  >
    <a
      aria-label="Link that will take you back to the home page of this blog wesbite."
      href="/"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    >
      Want us to take you back ‘home‘?
    </a>
  </div>
</div>
`;
